javaagent/build.gradle.kts

```
@@ -33,9 +36,19 @@ val copyAgent = tasks.register<Copy>("copyAgent") {
     rename("opentelemetry-javaagent-.*\\.jar", "opentelemetry-javaagent.jar")
 }
 
+val copyExtension = tasks.register<Copy>("copyExtension") {
+    from(extension.files) {
+        include("opentelemetry-samplers-*.jar")
+        exclude("*sources.jar")
+        exclude("*javadoc.jar")
+    }
```

I think this should work (see https://github.com/Azure-Samples/ApplicationInsights-Java-Samples/blob/e2d3255f34c779ed2c480d31fa8043ab8fcc449d/gradle-kotlin-dsl/build.gradle.kts#L25)
```suggestion
    from(extension.singleFile)
```

---

It doesn't work. I added some logging to see why:
```
agent.files.forEach(System.out::println)

extension.files.forEach(System.out::println)
```
Output for `agent`:
```
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry.javaagent/opentelemetry-javaagent/2.9.0/701e8c3e91802163e5e85f8c34326acea4c0bfba/opentelemetry-javaagent-2.9.0.jar
```

Output for `extension`:
```
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry.contrib/opentelemetry-samplers/1.40.0-alpha/5e3580e953fbf849adca836f69fdeebc2e50ebf7/opentelemetry-samplers-1.40.0-alpha.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry.semconv/opentelemetry-semconv/1.28.0-alpha/90d4d4dc28153eed4af75759b610a1bf81b77b75/opentelemetry-semconv-1.28.0-alpha.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-sdk/1.43.0/5e4e695beb5400c1866dcf800f08ca8307405b3b/opentelemetry-sdk-1.43.0.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-sdk-trace/1.43.0/69619c6433ebf0fe5196f7c3fe18bd543279b451/opentelemetry-sdk-trace-1.43.0.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-sdk-metrics/1.43.0/ca508a02ef7bb96678d054de264fcc0bf41e729c/opentelemetry-sdk-metrics-1.43.0.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-sdk-logs/1.43.0/e613c8ecf1efc1a275c7fe744773035df954c291/opentelemetry-sdk-logs-1.43.0.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-sdk-common/1.43.0/8f2bb94d9d9fe070de698b973c1f1f247a166dcf/opentelemetry-sdk-common-1.43.0.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-api-incubator/1.43.0-alpha/2dcee00ad7552d9190857e271f6066e655c6c719/opentelemetry-api-incubator-1.43.0-alpha.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-api/1.43.0/ce6889e709baac891833dc1c718702661650b3b3/opentelemetry-api-1.43.0.jar
/Users/jberg/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-context/1.43.0/473c7a67d2d587f7faefe7b6d9c00743e82819ee/opentelemetry-context-1.43.0.jar
```

Excluding transitive dependencies fixes the issue!

---

good to know, thanks 👍